R/utils.R

```diff
@@ -215,3 +214,19 @@ check_filter_datanames <- function(filters, datanames) {
     TRUE
   }
 }
+
+#' Wrapper on `teal.data::datanames`
+#'
+#' Special function used in internals of `teal` to return names of datasets even if `datanames`
+#' has not been set.
+#' @param data (`teal_data`)
+#' @return `character`
+#' @keywords internal
+teal_data_datanames <- function(data) {
+  checkmate::assert_class(data, "teal_data")
+  if (length(teal.data::datanames(data))) {
+    teal.data::datanames(data)
+  } else {
+    names(data@env)
```

Comment:
what about `ls(teal.code::get_env(data)` once this is merged https://github.com/insightsengineering/teal.code/pull/187

Reply:
`names` keeps hidden items.

Reply:
`ls(teal.code::get_env(data), all.names = TRUE)` : P ?